### PR TITLE
Created mutations for add, delete, and patch

### PR DESCRIPTION
**Pull request by vbusch**
Created mutations for Addresses & Address Spaces.  Implement Create, Patch, and Delete

Signed-off-by: Vanessa <vbusch@redhat.com>

### Type of change

- Bugfix
- Enhancement / new feature
- Refactoring
- Documentation

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: pkg/apis/enmasse/v1beta1/types.go [hunk 1]
```diff
@@ -206,7 +206,7 @@ type Address struct {
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
 	Spec   AddressSpec   `json:"spec"`
-	Status AddressStatus `json:"status,omitempty"`
+	Status *AddressStatus `json:"status,omitempty"`
```
Comment (k-wall) [inline pkg/apis/enmasse/v1beta1/types.go]: Why is this change necessary?   Having the status field a pointer isn't the common pattern for either our own objects (AddressSpace, AddressPlan etc), or for those of the Kubernetes core (Namespace, Pod etc).  Why deviate here?

File: cmd/console-server/main.go [hunk 2]
```diff
@@ -67,7 +67,7 @@ Major items that are unfinished:
 
 
 */
-func authHandler(next http.Handler) http.Handler {
+func authHandler(next http.Handler, coreClient *enmassev1beta1.EnmasseV1beta1Client) http.Handler {
```
Comment (k-wall) [inline cmd/console-server/main.go]: Passing the existing coreClient argument to this function is wrong.  The coreClient used for the mutations needs to be associated with the bearer token of the web user in order that the actions are carried out by the user (rather than the console-service service account(.  The authHandler needs to create its own coreClient.

File: pkg/apis/enmasse/v1beta1/types.go (hunk 1 above)
Comment (vbusch) [reply pkg/apis/enmasse/v1beta1/types.go]: I see this pattern a fair bit in k8s.io/api/core/v1/types.go, and occassionally in the current file.
type EndpointSpec struct {
	Name        string           `json:"name"`
	Service     string           `json:"service"`
	Certificate ***CertificateSpec** `json:"cert,omitempty"`
	Expose      ***ExposeSpec**      `json:"expose,omitempty"`
}

My golang knowledge isn't great.  But without using a pointer, it is difficult to tell the difference between an empty object and an unset object in go.  As a result, you end up with empty objects, instead of null objects.

When creating a new address we want the following to be sent:
```
apiVersion: enmasse.io/v1beta1
kind: Address
metadata:
    name: myspace.myqueue
spec:
    address: myqueue
    type: queue
    plan: standard-small-queue
```
But instead we end up with:
```
apiVersion: enmasse.io/v1beta1
kind: Address
metadata:
    name: myspace.myqueueess
spec:
    address: myqueueess
    type: queue
    plan: standard-small-queue
status:
   planStatus:
      resources: {}
```

Comment (lulf) [reply pkg/apis/enmasse/v1beta1/types.go]: Are we just missing an omitempty in the type annotation?

Comment (vbusch) [reply pkg/apis/enmasse/v1beta1/types.go]: Address type already had the omitempty for AddressStatus
```
type Address struct {
	metav1.TypeMeta   `json:",inline"`
	metav1.ObjectMeta `json:"metadata,omitempty"`

	Spec   AddressSpec   `json:"spec"`
	Status AddressStatus `json:"status,omitempty"`
}
```
Is there another type, that I should look at?

Comment (lulf) [reply pkg/apis/enmasse/v1beta1/types.go]: I think this is fixed on master now @vbusch by making the planStatus nullable. Actually, the same as your second change in this PR. So you can remove the pointer on AddressStatus I think, rebase  on master, and it should hopefully work.

File: pkg/apis/enmasse/v1beta1/types.go [hunk 3]
```diff
@@ -236,7 +236,7 @@ type AddressStatus struct {
 	Phase          string            `json:"phase,omitempty"`
 	Messages       []string          `json:"messages,omitempty"`
 	BrokerStatuses []BrokerStatus    `json:"brokerStatus,omitempty"`
-	PlanStatus     AddressPlanStatus `json:"planStatus,omitempty"`
+	PlanStatus     *AddressPlanStatus `json:"planStatus,omitempty"`
```
[the remark this answers is not in the document]
Comment (lulf) [reply pkg/apis/enmasse/v1beta1/types.go]: I made the same change in #3689

**Comment by k-wall** [comment]
As @lulf suggested, I rebased console-server against master.

**Comment by k-wall** [comment]
@vbusch I took the liberty of rebasing your branch and incorporating the change to user the logged on user's coreClient
